src/rich-text-editor.component.js

```diff
@@ -142,12 +142,19 @@ export const RichTextEditor = forwardRef((props, editorRef) => {
   useEffect(() => {
     if (props.placeholder) {
       const styleElement = document.createElement('style')
-      styleElement.textContent = `.bandicoot-id-${bandicootId.current}:empty:before { content: attr(data-placeholder); ${getBrowserDefaultStyling()} }`
+      styleElement.textContent = `.bandicoot-id-${bandicootId.current}:empty:before { content: attr(data-placeholder); }`
       document.head.appendChild(styleElement)
 
+      const styleSheet = Array.prototype.slice.call(document.styleSheets).find(s => s.ownerNode === styleElement)
+      const styleDeclaration = styleSheet.cssRules[0].style
+      const styles = props.placeholderStyle ? props.placeholderStyle : getDefaultPlaceholderStyles()
```

Comment:
`styles` is sometimes an object and sometimes a string. However your code always treats it as an object (by iterating over its keys). This should be fixed by making getDefaultPlaceholderStyles return an object instead of string.